infra/build/functions/sync/main.py

```diff
@@ -35,41 +47,141 @@ class GitAuth(ndb.Model):
+  project_id = os.environ.get('GCP_PROJECT')
+  location_id = os.environ.get('FUNCTION_REGION')
+  parent = cloud_scheduler_client.location_path(project_id, location_id)
+  job = {
+      'name': parent + '/jobs/' + project.name + '-scheduler',
+      'pubsub_target': {
+          'topic_name': 'projects/' + project_id + '/topics/request-build',
+          'data': project.name.encode()
+      },
+      'schedule': project.schedule
+  }
+
+  update_mask = {'schedule': schedule}
+  cloud_scheduler_client.update(job, update_mask)
+
+
+def sync_projects(cloud_scheduler_client, projects):
   """Sync projects with cloud datastore."""
   project_query = Project.query()
-  projects_to_remove = [
-      project.key for project in project_query if project.name not in projects
-  ]
+  for project in project_query:
+    if project.name in projects:
+      continue
 
-  ndb.delete_multi(projects_to_remove)
+    try:
+      delete_scheduler(cloud_scheduler_client, project.name)
+      project.key.delete()
+    except exceptions.GoogleAPICallError as error:
+      logging.error('Scheduler deletion for %s failed with %s', project.name,
+                    error)
 
   existing_projects = {project.name for project in project_query}
 
-  new_projects = [
-      Project(name=project)
-      for project in projects
-      if project not in existing_projects
-  ]
-  ndb.put_multi(new_projects)
+  for project_name in projects:
+    if project_name in existing_projects:
+      continue
+
+    try:
+      create_scheduler(cloud_scheduler_client, project_name,
+                       projects[project_name])
+      Project(name=project_name, schedule=projects[project_name]).put()
+    except exceptions.GoogleAPICallError as error:
+      logging.error('Scheduler creation for %s failed with %s', project_name,
+                    error)
+
+  for project in project_query:
+    if project.name not in projects or project.schedule == projects[
+        project.name]:
+      continue
+    try:
+      update_scheduler(cloud_scheduler_client, project, projects[project.name])
+      project.schedule = projects[project.name]
+      project.put()
+    except exceptions.GoogleAPICallError as error:
+      logging.error('Updating scheduler for %s failed with %s', project.name,
+                    error)
+
+
+def _has_docker_file(project_contents):
+  """Checks if project has a Dockerfile."""
+  return any(
+      content_file.name == 'Dockerfile' for content_file in project_contents)
 
 
-def _has_docker_file(repo, project_path):
-  """Checks if project has a Dockerfile."""
-  return any(content_file.name == 'Dockerfile'
-             for content_file in repo.get_contents(project_path))
+def get_schedule(project_contents):
+  """Checks for schedule parameter in yaml file else uses DEFAULT_SCHEDULE."""
+  for content_file in project_contents:
+    if content_file.name != 'project.yaml':
+      continue
+    yaml_str = content_file.decoded_content.decode('utf-8')
```

Comment:
nit: no need for `yaml_str` variable. just pass the value straight into yaml.safe_load.